src/hooks/useCompanyProfile.ts

```diff
@@ -0,0 +1,81 @@
+import { useQuery } from '@tanstack/react-query';
+import {
+  CompanyProfile,
+  CompanyProfileArraySchema,
+} from '@/components/AddTransactionForm/companyProfileSchema';
+import {
+  getCompanyProfileFromDB,
+  saveCompanyProfileToDB,
+} from '@/lib/companyProfileIDB';
+import { api } from '@/api/client';
+
+const FALLBACK_LOGO = 'src/assets/fallback_logo.png';
+
+const validateImageUrl = async (url: string | null): Promise<string> => {
+  try {
+    if (!url) {
+      return FALLBACK_LOGO;
+    }
```

Comment:
Here some responsibilities are mixed. This function should be responsible only for validation, that returns `true` or `false`. Also it should not accept `null` as an input. If the input data should be `null` we don't need to validate.